src/pages/WorkspaceDetails/EditorTab/index.tsx

```diff
@@ -266,6 +285,13 @@ export class EditorTab extends React.PureComponent<Props, State> {
       return;
     }
     const workspaceCopy = convertWorkspace(this.props.workspace.ref);
+    if (!devfile.metadata) {
+      devfile.metadata = {};
+    }
+    if (!devfile.metadata.name) {
+      devfile.metadata.name = workspaceCopy.name;
```

Comment:
it would be cool if we recover namespace for devworkspace as well
![Screenshot_20210617_150559](https://user-images.githubusercontent.com/5887312/122393702-d36e1280-cf7d-11eb-881f-7cbf06a72857.png)

Reply:
done!